### PR TITLE
Fix the issues that devops credentials cannot be deleted

### LINKED ISSUE #3353: The credential can not be deleted.
**Describe the Bug**
image
image

**Environment**
http://allinone:30880

**Preset conditions**
There is a devops 'dev', it has credential named '6'

**How To Reproduce**
1. Go to devops 'dev'
2. Click on  'Credentials'
3. Delete credential '6'

**Expected behavior**
The credential was deleted successfully.

/priority high
/area devops
/cc @kubesphere/sig-devops
/kind bug
/milestone 3.1.0

### LINKED ISSUE #3355: Devops's namespace can't be deleted
**Describe the Bug**
The Namespace that created by a Devops remains **Terminating** status if there is a credential was created.

**Versions Used**
KubeSphere: V3.0.0

**How To Reproduce**
1. Create a Devops Project under the Workspace.
2. Create a new 'Account' credential under the Devops project.
3. Delete Devops project 
4. Get namespace with command `kubectl get namespace devopsxxx -o yaml`

```bash
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    kubesphere.io/creator: admin
  creationTimestamp: "2021-02-09T07:20:45Z"
  deletionTimestamp: "2021-02-09T07:22:01Z"
  labels:
    kubesphere.io/devopsproject: wsp-deleteopspfxfn
    kubesphere.io/namespace: wsp-deleteopspfxfn
spec:
  finalizers:
  - kubernetes
status:
  conditions:
  - lastTransitionTime: "2021-02-09T07:22:07Z"
    message: All resources successfully discovered
    reason: ResourcesDiscovered
    status: "False"
    type: NamespaceDeletionDiscoveryFailure
  - lastTransitionTime: "2021-02-09T07:22:07Z"
    message: All legacy kube types successfully parsed
    reason: ParsedGroupVersions
    status: "False"
    type: NamespaceDeletionGroupVersionParsingFailure
  - lastTransitionTime: "2021-02-09T07:22:07Z"
    message: All content successfully deleted, may be waiting on finalization
    reason: ContentDeleted
    status: "False"
    type: NamespaceDeletionContentFailure
  - lastTransitionTime: "2021-02-09T07:22:07Z"
    message: 'Some resources are remaining: secrets. has 1 resource instances'
    reason: SomeResourcesRemain
    status: "True"
    type: NamespaceContentRemaining
  - lastTransitionTime: "2021-02-09T07:22:07Z"
    message: 'Some content in the namespace has finalizers remaining: finalizers.kubesphere.io/credential
      in 1 resource instances'
    reason: SomeFinalizersRemain
    status: "True"
    type: NamespaceFinalizersRemaining
  phase: Terminating
```

**Expected behavior**
The namespace can be deleted successful.

@kubesphere/developers

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind bug
/area devops

**What this PR does / why we need it**:
DevOps credential cannot be delete or updated.

**Which issue(s) this PR fixes**:
Fixes #3353
fixes #3355

**Special notes for reviewers**:
None.

**Additional documentation, usage docs, etc.**:
None.

### CODE REVIEW

File: pkg/apis/devops/v1alpha3/devopsproject_types.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ const (
 	DevOpeProjectSyncStatusAnnoKey = DevOpsProjectPrefix + "syncstatus"
 	DevOpeProjectSyncTimeAnnoKey   = DevOpsProjectPrefix + "synctime"
 	DevOpeProjectSyncMsgAnnoKey    = DevOpsProjectPrefix + "syncmsg"
+	DevOpsCredentialDataHash       = "devopscredential.devops.kubesphere.io/datahash"
```
Comment (shaowenchen) [inline pkg/apis/devops/v1alpha3/devopsproject_types.go:36]: DevOpsCredentialDataHash should not be defined in devopsproject_types.go, but in credential_types.go. 

Please use DevOpsCredentialPrefix  + datahash replace "devopscredential.devops.kubesphere.io/datahash".

Keeping style consistent is more important than function.

**Review by shaowenchen — COMMENTED** [review]
@wenxin-01  Please pay attention to this pr.

File: pkg/apis/devops/v1alpha3/devopsproject_types.go (hunk 1 above)
Comment (LinuxSuRen) [reply pkg/apis/devops/v1alpha3/devopsproject_types.go:36]: Sure.

**Comment by LinuxSuRen** [comment]
This PR will close #3355 as well.

**Pull request merged**